mixer/adapter/kubernetesenv/cache.go

```diff
@@ -39,228 +35,70 @@ type (
 	}
 
 	controllerImpl struct {
-		clientset     kubernetes.Interface
-		env           adapter.Env
-		pods          cache.SharedInformer
-		mutationsChan chan resourceMutation
-
-		ipPodMap      map[string]string
-		ipPodMapMutex *sync.RWMutex
-	}
-
-	// used to send updates to the logger
-	resourceMutation struct {
-		kind eventType
-		obj  interface{}
+		cache.SharedIndexInformer
 	}
-
-	eventType int
-)
-
-const (
-	addition eventType = iota
-	update
-	deletion
 )
 
-// mutationBufferSize sets the limit on how many mutation events can be
-// outstanding at any moment. 100 is chosen as a reasonable default to start.
-// TODO: make this configurable
-const mutationBufferSize = 100
-
-// errorDelay controls how long the logger waits when encountering an error
-// during logging. This should only ever happen when the underlying cache has
-// not yet synced (at which point we need to wait before doing any further
-// processing).
-// TODO: make this configurable
-const errorDelay = 1 * time.Second
-
-const debugVerbosityLevel = 4
+func ipIndex(obj interface{}) ([]string, error) {
+	pod, ok := obj.(*v1.Pod)
+	if !ok {
+		return nil, errors.New("object is not a pod")
+	}
+	ip := pod.Status.PodIP
+	if ip == "" {
+		return nil, nil
```

Comment:
should this be `return []string{}, nil` or is nil expected?

Reply:
`nil` is actually okay and safe, as it behaves like an empty slice.